vcd/resource_vcd_vapp_vm.go

```diff
@@ -583,421 +591,739 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
+			Name:             vmName, // VM name post creation
+			PowerOn:          false,  // VM will be powered on after all configuration is done
+			AllEULAsAccepted: d.Get("accept_all_eulas").(bool),
+			ComputePolicy:    vmComputePolicy,
+			SourcedVmTemplateItem: &types.SourcedVmTemplateParams{
+				Source: &types.Reference{
+					HREF: vmTemplate.VAppTemplate.HREF,
+					ID:   vmTemplate.VAppTemplate.ID,
+					Type: vmTemplate.VAppTemplate.Type,
+					Name: vmTemplate.VAppTemplate.Name,
+				},
+				VmGeneralParams: &types.VMGeneralParams{
+					Description: d.Get("description").(string),
+				},
+				VmTemplateInstantiationParams: &types.InstantiationParams{
+					// If a MAC address is specified for NIC - it does not get set with this call,
+					// therefore an additional `vm.UpdateNetworkConnectionSection` is required.
+					NetworkConnectionSection: &networkConnectionSection,
+				},
+				StorageProfile: storageProfilePtr,
+			},
 		}
 
-		// TODO do not trigger resourceVcdVAppVmUpdate from create. These must be separate actions.
-		err = resourceVcdVAppVmUpdateExecute(d, meta, "create", vmType, sizingPolicy)
-		if err != nil {
-			errAttachedDisk := updateStateOfAttachedDisks(d, *vm)
-			if errAttachedDisk != nil {
-				dSet(d, "disk", nil)
-				return fmt.Errorf("error reading attached disks : %s and internal error : %s", errAttachedDisk, err)
-			}
-			return err
-		}
-	} else {
-		//create empty VM
-		vm, err := addEmptyVm(d, vcdClient, org, vdc, vappName)
+		util.Logger.Printf("%# v", pretty.Formatter(standaloneVmParams))
+		vm, err = vdc.CreateStandaloneVMFromTemplate(&standaloneVmParams)
 		if err != nil {
 			d.SetId("")
-			return fmt.Errorf("[VM creation] error creating standalone VM %s : %s", vmName, err)
+			return nil, fmt.Errorf("[VM creation] error creating standalone VM from template %s : %s", vmName, err)
 		}
-		util.Logger.Printf("[VM create] VM after creation %# v", pretty.Formatter(vm.VM))
+
+		d.SetId(vm.VM.ID)
+
+		util.Logger.Printf("[VM create] VM from template after creation %# v", pretty.Formatter(vm.VM))
 		vapp, err = vm.GetParentVApp()
 		if err != nil {
 			d.SetId("")
-			return fmt.Errorf("[VM creation] error retrieving vApp from standalone VM %s : %s", vmName, err)
+			return nil, fmt.Errorf("[VM creation] error retrieving vApp from standalone VM %s : %s", vmName, err)
 		}
 		util.Logger.Printf("[VM create] vApp after creation %# v", pretty.Formatter(vapp.VApp))
 		dSet(d, "vapp_name", vapp.VApp.Name)
+		dSet(d, "vm_type", string(standaloneVmType))
 
-		err = updateAdvancedComputeSettings(d, vm)
+		////////////////////////////////////////////////////////////////////////////////////////////
+		// This part of code handles additional VM create operations, which can not be set during
+		// initial VM creation.
+		// __Explicitly__ template based Standalone VMs are addressed here.
+		////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Just in case: is this comment block correctly placed? It shows up as the last and indented differently (but it may just be how GitHub renders).

Reply:
Yes. This often happens with `switch/case` code. I have moved it left by 1 tab. Also noticed that similar comments were missing in a few other places - added them.